src/ast/visitor.rs

```diff
@@ -76,10 +76,12 @@ pub trait NodeVisitor<'a> {
 
     fn super_cond(
         &mut self,
-        if_blk: &Located<Block<'a>>,
-        do_blk: &Located<Block<'a>>,
+        branch: &Branch<'a>,
         el_blk: &Located<Block<'a>>,
     ) {
+        let if_blk = &branch.cond;
```

Comment:
Could you add a `visit_branch` and `super_branch` methods to the trait? I think that'd be easier to maintain.